core/src/main/scala/com/mesosphere/usi/core/SchedulerLogicHandler.scala

```diff
@@ -118,15 +118,38 @@ private[core] class SchedulerLogicHandler(mesosCallFactory: MesosCalls) {
     frameResultBuilder.result
   }
 
+  def generateSuppressCalls(pendingLaunch: Set[PodId], newLaunched: Set[PodId]): List[Call] = {
+    val alreadyLaunchedRoles = newLaunched.iterator.collect {
+      case id if specs.podSpecs.contains(id) => specs.podSpecs(id).runSpec.role
+    }
+
+    val rolesBeingLaunched = pendingLaunch
+      .map(id => specs.podSpecs.get(id))
+      .collect { case Some(p) => p.runSpec.role }
+
+    alreadyLaunchedRoles
+      .filterNot(rolesBeingLaunched)
+      .map { r =>
+        mesosCallFactory.newSuppress(Some(r))
+      }
+      .toList
+  }
+
   private def updateCachesAndRevive(
       specs: SpecState,
       state: SchedulerState,
       dirtyPodIds: Set[PodId]): SchedulerEvents = {
-    this.cachedPendingLaunch = this.cachedPendingLaunch.update(specs, state, dirtyPodIds)
-    if (cachedPendingLaunch.pendingLaunch.nonEmpty)
-      SchedulerEvents(mesosCalls = List(mesosCallFactory.newRevive(None)))
-    else
-      SchedulerEvents.empty
+    val updateResult = this.cachedPendingLaunch.update(specs, state, dirtyPodIds)
+    this.cachedPendingLaunch = updateResult.newCachedPendingLaunch
+    val reviveCalls = updateResult.newToBeLaunched.iterator
+      .map(id => specs.podSpecs.get(id))
+      .collect { case Some(p) => p.runSpec.role }
```

Comment:
I am wondering if we need to do this since we're going to be debouncing at later stages... but I'll do it! We'll need to allocate some extra collections because of that though...